### PR TITLE
TACKLE-611: Cascade assessment delete when app is deleted

### PR DESCRIPTION
**Pull request by ibolton336**
https://issues.redhat.com/projects/TACKLE/issues/TACKLE-611

### CODE REVIEW

**Comment by mturley** [comment]
What drives the underlying storage of these applications/assessments? They aren't k8s CRs, right? I wonder if this API has a mechanism similar to k8s `ownerReferences`, so the assessments can be created with metadata saying they are owned by the application and have the backend auto-delete them when the app is deleted.

**Comment by ibolton336** [comment]
> What drives the underlying storage of these applications/assessments? They aren't k8s CRs, right? I wonder if this API has a mechanism similar to k8s `ownerReferences`, so the assessments can be created with metadata saying they are owned by the application and have the backend auto-delete them when the app is deleted. That way the UI doesn't have to be concerned with these relationships.

Good question - I think there is a backend golang mechanism to do just that. The issue here is that these requests span across multiple APIs & the solution is not fully fleshed out on the backend yet so Jeff requested that we do this temporarily on the UI side. See comments https://issues.redhat.com/projects/TACKLE/issues/TACKLE-611

**Comment by mturley** [comment]
Got ya, ok. Thanks.

File: pkg/client/src/app/queries/applications.ts [hunk 1]
```diff
@@ -113,13 +109,19 @@ export const useCreateApplicationMutation = (
...
 ) => {
   const queryClient = useQueryClient();
-  return useMutation(deleteApplication, {
-    onSuccess: () => {
+  const { mutate: mutateAssessments } = useDeleteAssessmentMutation();
+  return useMutation(({ id }: { id: number }) => deleteApplication(id), {
+    onSuccess: (_, { id }) => {
       onSuccess();
       queryClient.invalidateQueries(ApplicationsQueryKey);
+      const assessment = getApplicationAssessment(id);
```
Comment (mturley) [inline pkg/client/src/app/queries/applications.ts:121]: @ibolton336 When I try to test this, I'm getting an error on this line when I delete an application:
```
TypeError: getApplicationAssessment is not a function
    at Object.onSuccess (applications.ts?33ea:121:26)
    at eval (mutation.js?832d:92:1)
```

Comment (ibolton336) [reply pkg/client/src/app/queries/applications.ts:121]: Looks like i only fixed it for the assessments page & this function is reused in the analysis page also. Fixing @mturley

**Review by gildub — COMMENTED** [review]
LGTM.

**Review by gildub — APPROVED** [review]

**Pull request merged**